tools/zapdir/main.go

```diff
@@ -0,0 +1,58 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/Microsoft/hcsshim/internal/wclayer"
+	"github.com/urfave/cli"
+)
+
+const (
+	dirArgName = "dir"
+)
+
+func main() {
+	app := cli.NewApp()
+	app.Name = "zapdir"
+	app.Usage = "Delete a directory"
+
+	app.Flags = []cli.Flag{
+		cli.StringFlag{
+			Name:  dirArgName,
+			Value: "",
+			Usage: "Directory to delete",
+		},
+	}
+
+	app.Action = func(c *cli.Context) error {
+		dir := c.String(dirArgName)
+
+		if dir == "" {
+			return errors.New("dir must be supplied")
+		}
+
+		// DestroyLayer requires an absolute path.
+		dir, err := filepath.Abs(dir)
+		if err != nil {
+			return err
+		}
+
+		if _, err := os.Stat(dir); os.IsNotExist(err) {
```

Comment:
I don't know but if `os.Stat` can fail for any reason other than `IsNotExit` we will silently ignore here. We should likely stop since something seems very wrong in this case.

Reply:
This was the same behavior as in docker-ci-zap, so i thought it might be expected, but I'm happy to not ignore this error. :)